File: java-solutions/src/main/java/org/karabalin/s1929/Solution1929.java
package org.karabalin.s1929;

import org.junit.jupiter.api.Assertions;

class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int[] result = new int[nums.length * 2];
        for (int i = 0; i < nums.length; i++) {
            result[i] = nums[i];
            result[i + n] = nums[i];
        }
        return result;
    }
}

public class Solution1929 {
    public static void main(String[] args) {
        var s = new Solution();
        Assertions.assertArrayEquals(new int[]{1, 2, 1, 1, 2, 1}, s.getConcatenation(new int[]{1, 2, 1}));
    }
}

File: kotlin/src/main/kotlin/org/karabalin/s1929/Solution1929.kt
package org.karabalin.s1929

class Solution {
    fun getConcatenation(nums: IntArray): IntArray {
        val n = nums.size
        val result = IntArray(n * 2)
        for (i in nums.indices) {
            result[i] = nums[i]
            result[i + n] = nums[i]
        }
        return result
    }
}

fun main() {
    val solution = Solution()
    val result = solution.getConcatenation(intArrayOf(1, 2, 1))
    println(result.joinToString())
}
